DigitalLearningSolutions.Data.Tests/Services/TrackerServiceTests.cs

```diff
@@ -50,19 +52,55 @@ public void ProcessQuery_with_unknown_action_returns_InvalidAction_response()
         }
 
         [Test]
-        public void
-            ProcessQuery_with_GetObjectiveArray_action_passes_query_params_and_returns_appropriate_service_response()
+        public void ProcessQuery_with_GetObjectiveArray_action_passes_query_params()
         {
             // Given
             var query = new TrackerEndpointQueryParams
                 { Action = "GetObjectiveArray", CustomisationId = 1, SectionId = 1 };
-            A.CallTo(() => actionService.GetObjectiveArray(1, 1)).Returns("GOA result");
+
+            // When
+            trackerService.ProcessQuery(query);
+
+            // Then
+            A.CallTo(() => actionService.GetObjectiveArray(1, 1)).MustHaveHappenedOnceExactly();
+        }
+
+        [Test]
+        public void ProcessQuery_with_GetObjectiveArray_action_correctly_serialises_contentful_response()
+        {
+            // Given
+            var dataToReturn = new GetObjectiveArrayData(new[] { new Objective(1, new List<int>{6,7,8}, 4), new Objective(2, new List<int>{17,18,19}, 0) });
+            var expectedJson =
+                "{\"objectives\":[{\"tutorialid\":1,\"interactions\":[6,7,8],\"possible\":4,\"myscore\":0}," +
+                "{\"tutorialid\":2,\"interactions\":[17,18,19],\"possible\":0,\"myscore\":0}]}";
+
+            var query = new TrackerEndpointQueryParams
+                { Action = "GetObjectiveArray", CustomisationId = 1, SectionId = 1 };
+            A.CallTo(() => actionService.GetObjectiveArray(1, 1)).Returns(dataToReturn);
+
+            // When
+            var result = trackerService.ProcessQuery(query);
+
+            // Then
+            result.Should().Be(expectedJson);
+        }
+
+        [Test]
+        public void ProcessQuery_with_GetObjectiveArray_action_correctly_serialises_null_response()
+        {
+            // Given
+            GetObjectiveArrayData? dataToReturn = null;
+            var expectedJson = "{}";

```

Comment:
nitpick - I'd just inline `expectedJson`. To see what's expected I'd normally look at the `Then` section rather than the `Given` section.